src/server.js

```diff
@@ -277,8 +278,13 @@ function Server(host, port, key, cert)
 
     this.resetHandlers = function() {
         handlers = [];
+        self.cleanRequests();
```

Comment:
I'm not 100% sure in cleaning the history of the requests inside an already existing method called `resetHandlers`, for a couple of reasons:
1 - we are linking cleaning a configuration with getting rid of all the traces left, that could be yet useful
2 - we are changing the behaviour of a method that could potentially breaks past implementations (I'm thinking if someone are using the `resetHandlers` method before doing expectations about requests)

I think we should add another method `reset()` where trigger both `resetHandlers` and `cleanRequests` methods. What do you think @loicsimon and @lele85?

Reply:
Yep @teorossi82, probably having the three methods like this:

- `reset()`
- `resetHandlers()`
- `resetRequests()`

where `reset()` is just `resetHandlers(); resetRequests()` is the best way to go without changing behaviour of existing APIs.

@loicsimon what is your take?

Reply:
You're right

I don't want to introduce breaking changes for other users.
I'll do my best to update this PR by tomorrow.

Reply:
Thanks!